packages/@lwc/integration-karma/test/api/registerStylesheet/index.spec.js

```diff
@@ -0,0 +1,60 @@
+import {
+    createElement,
+    LightningElement,
+    registerComponent,
+    registerStylesheet,
+    registerTemplate,
+} from 'lwc';
+import Test from 'x/test';
+
+it('should accept a function and return the same value', () => {
+    const stylesheet = () => '';
+    const result = registerStylesheet(stylesheet);
+
+    expect(result).toBe(stylesheet);
+});
+
+it('should throw if a component tries to use a stylesheet that is not registered', () => {
+    const stylesheet = () => '';
+    function tmpl() {
+        return [];
+    }
+    tmpl.stylesheetToken = 'x-component_component';
+    tmpl.stylesheets = [stylesheet];
+    registerTemplate(tmpl);
+    class CustomElement extends LightningElement {}
+    registerComponent(CustomElement, { tmpl });
+
+    const elm = createElement('x-component', { is: CustomElement });
+
+    expect(() => {
+        document.body.appendChild(elm);
+    }).toThrowError(TypeError, 'Unexpected LWC stylesheet content.');
+});
+
+it('should not throw if a component tries to use a stylesheet that is registered', () => {
+    const stylesheet = () => '';
+    function tmpl() {
+        return [];
+    }
+    tmpl.stylesheetToken = 'x-component_component';
+    tmpl.stylesheets = [stylesheet];
+    registerStylesheet(stylesheet);
+    registerTemplate(tmpl);
+
+    class CustomElement extends LightningElement {}
+    registerComponent(CustomElement, { tmpl });
+
+    const elm = createElement('x-component', { is: CustomElement });
+
+    expect(() => {
+        document.body.appendChild(elm);
+    }).not.toThrow();
+});
+
+it('should throw if registerStylesheet is used from userland code', () => {
```

Comment:
This test currently fails; where can I find the mechanism that would protect this function form use in userland code?

Reply:
I think for `registerTemplate` and `registerComponent`, these are already protected by [the ESLint rules](https://github.com/salesforce/eslint-plugin-lwc/blob/618a4aeb2ea0275e0ab50fc32ef1ec641234c9f6/lib/rules/no-disallowed-lwc-imports.js#L11-L35). There's nothing you need to do, since it's allowlist-based.